mod-source-record-manager-server/src/main/java/org/folio/verticle/AbstractConsumersVerticle.java

```diff
@@ -54,17 +56,20 @@ public void start(Promise<Void> startPromise) {
         .subscriptionDefinition(subscriptionDefinition)
         .build());
     });
+    List<Future> futures = new ArrayList<>();
+    consumerWrappersList.forEach(consumerWrapper ->
+      futures.add(consumerWrapper.start(getHandler(), PubSubClientUtils.constructModuleName())));
 
-    consumerWrappersList.forEach(consumerWrapper -> consumerWrapper
-      .start(getHandler(), PubSubClientUtils.constructModuleName())
-      .onComplete(ar -> startPromise.complete()));
+    CompositeFuture.all(futures).onComplete(ar -> startPromise.complete());
   }
 
   @Override
   public void stop(Promise<Void> stopPromise) {
-    consumerWrappersList.forEach(consumerWrapper -> consumerWrapper
-      .stop()
-      .onComplete(ar -> stopPromise.complete()));
+    List<Future> futures = new ArrayList<>();
+    consumerWrappersList.forEach(consumerWrapper ->
+      futures.add(consumerWrapper.stop()));
+
+    CompositeFuture.all(futures).onComplete(ar -> stopPromise.complete());
```

Comment:
```suggestion
    CompositeFuture.join(futures).onComplete(ar -> stopPromise.complete());
```
suggest to make sure that all consumers have been stopped.
Thank you, Kate!